### DISCUSSION TITLE
Meross truly Local?

### DISCUSSION THREAD

**Discussion by yestrdave** [krahabb/meross_lan#322] [Q&A]
I have been playing around most of the day with two additional Meross garage door openers I installed and my Mothers house. I have 6 using Meross LAN at my house. Thank you so much for the app! May convert to the below....

Here are a couple questions. I originally installed using ip address and key in HA using the Meross app.   I figured if I could delete the opener out of the Meross app and I would be truly local.  Here is what I did.  I deleted both openers from the Meross app. I did nothing in HA. Left the Meross LAN configured.  Installed in HK app using HK code. Appeared in HK and not in the Meross App.  Then I deleted out of HK and installed it with a new name in HA under the HK Devices integration.  At this point I'm thinking I have no links back to Meross.   Surprisingly, the Meross LAN configuration using the ip address and key started to work after all this.   Could the openers still be going back to Meross or is there an explanation in the Meross LAN integration?

I have confirmed with the internet disconnected Meross LAN works. 

This must be done in this order

Install Meross using app and setup Meross Lan. Make sure everything works. 

Leave everything in place within HA Meross lan. Delete device from Meross app. 

Add meross device to HK. Then delete it.  

The device should show up in HA.  Then install the device in HA using the HA devices. 

Once installed this way Meross LAN should begin working again without the need for internet

**Comment by krahabb** [comment]
Hello @yestrdave,
Thank you for your 'quick guide' on how to make your devices 'local': this could be useful for someone else too!

`meross_lan` in it's essence is designed to be fully local (i.e. almost all of the functionalities are implemented and supported locally) by directly communicating to the devices over HTTP.
There are also features to support MQTT (this implies connecting to a broker-server) with the same functionalities: basically, the devices support all of their features on different protocols: 
- HTTP (direct device connection) 
- MQTT (broker connection)
- HOMEKIT (another local protocol allowing direct connection to the device)
- MATTER (similar to HK - just newer and more open)

The Meross cloud services work by configuring the devices to use MQTT to a Meross owned broker. If you 'kill' this layer you're basically cutting the device out of the cloud

So yes, `meross_lan` is able to locally talk via HTTP to the devices provided the IP address and the device key. If you want to be sure there is no 'unsolicited traffic' you can set the protocol behavior in `meross_lan` to a fixed HTTP mode (AUTO will always try to use the best available protocol so, depending on the overall configuration, it could automatically try using MQTT to a Meross broker in order to connect to the devices)

**Comment by yestrdave** [comment]
Thanks for the reply @krahabb  I tried the http setting and I seam to be getting multiple errors and warnings. This one is not in the Meross App anymore.  It is also installed as a HK device

Logger: custom_components.meross_lan
Source: custom_components/meross_lan/helpers.py:479
Integration: Meross LAN (documentation, issues)
First occurred: 7:28:01 AM (2 occurrences)
Last logged: 7:28:31 AM

Logger: custom_components.meross_lan
Source: custom_components/meross_lan/helpers.py:501
Integration: Meross LAN (documentation, issues)
First occurred: 7:28:01 AM (2 occurrences)
Last logged: 7:28:31 AM

Logger: custom_components.meross_lan
Source: custom_components/meross_lan/helpers.py:458
Integration: Meross LAN (documentation, issues)
First occurred: 7:28:01 AM (2 occurrences)
Last logged: 7:28:31 AM


I got over 700 yesterday.  I deleted this device out of Meross LAN after that and was using HK only.  ( for only this device all others are still installed with Meross LAN and also show in the Meross AP.)  I installed it again after your comment and above is what I got.

I’ve been using this integration since you developed it.
